control-plane/src/modules/deployment/lambda.ts

```diff
@@ -0,0 +1,98 @@
+import { ZodSchema, z } from "zod";
+import {
+  CreateFunctionCommand,
+  CreateFunctionCommandOutput,
+  CreateFunctionRequest,
+  LambdaClient,
+  UpdateFunctionCodeCommand,
+} from "@aws-sdk/client-lambda";
+import { Deployment, s3AssetDetails } from "./deployment";
+import { DeploymentProvider, fetchConfig } from "./deployment-provider";
+
+type LambdaProviderConfig = Pick<
+  CreateFunctionRequest,
+  "Role" | "Handler" | "Runtime" | "Timeout"
+>;
+
+export class LambdaProvider implements DeploymentProvider {
+  public name(): string {
+    return "lambda";
+  }
+
+  public schema(): ZodSchema<LambdaProviderConfig> {
+    return z.object({
+      Role: z.string(),
+      Handler: z.string().default("index.handler"),
+      Runtime: z.enum(["nodejs20.x"]).default("nodejs20.x"),
+      Timeout: z.number().default(60),
+    });
+  }
+
+  public async createDeployment(
+    deployment: Deployment,
+  ): Promise<CreateFunctionCommandOutput> {
+    const config = await this.config();
+    const functionName = this.buildFunctionName(deployment);
+    const lambdaClient = new LambdaClient();
+
+    console.log("Creating new lambda", functionName);
+
+    try {
+      return await lambdaClient.send(
+        new CreateFunctionCommand({
+          ...config,
+          Tags: {
+            clusterId: deployment.clusterId,
+            service: deployment.service,
+          },
+          Publish: true,
+          FunctionName: functionName,
+          Code: {
+            ...s3AssetDetails(deployment),
+          },
+        }),
+      );
+    } catch (error: any) {
+      if (error.name === "ResourceConflictException") {
+        throw new Error(
+          `${functionName} service has already been created. It should be updated instead.`,
```

Comment:
These errors would indicate that our internal state (`deployment.status`) is not correct. We could _fallback_ to the `updateDeployment`.